apps/spruce/src/pages/projectAndRepoSettings/ProjectSettings.tsx

```diff
@@ -61,39 +63,51 @@ const ProjectSettings: React.FC = () => {
     },
   });
 
-  const { data: projectData, loading: projectLoading } = useQuery<
-    ProjectSettingsQuery,
-    ProjectSettingsQueryVariables
-  >(PROJECT_SETTINGS, {
-    skip: identifierIsObjectId || !projectIdentifier,
-    variables: { projectIdentifier },
-    onCompleted: (data) => {
-      if (data?.projectSettings?.projectRef?.hidden) {
-        dispatchToast.error(`Project is hidden.`);
-      }
-    },
-    onError: (e) => {
-      dispatchToast.error(
-        `There was an error loading the project ${projectIdentifier}: ${e.message}`,
-      );
+  const {
+    data: projectData,
+    error: projectError,
+    loading: projectLoading,
+  } = useQuery<ProjectSettingsQuery, ProjectSettingsQueryVariables>(
+    PROJECT_SETTINGS,
+    {
+      skip: identifierIsObjectId || !projectIdentifier,
+      variables: { projectIdentifier },
     },
-  });
+  );
+  useErrorToast(
+    projectError,
+    `There was an error loading the project ${projectIdentifier}`,
+  );
+
+  // Show error toast if project is hidden
+  const hasShownHiddenToast = useRef(false);
+  useEffect(() => {
+    // Reset ref when project changes
+    hasShownHiddenToast.current = false;
+  }, [projectIdentifier]);
+
+  useEffect(() => {
+    if (
+      projectData?.projectSettings?.projectRef?.hidden &&
+      !hasShownHiddenToast.current
+    ) {
+      hasShownHiddenToast.current = true;
+      dispatchToast.error(`Project is hidden.`);
+    }
+  }, [projectData?.projectSettings?.projectRef?.hidden, dispatchToast]);
```

Comment:
I kind of wonder if `useErrorToast` could be expanded to handle this case. e.g.

```suggestion
  // toast for hidden project
  useErrorToast(
    new Error("My error message"),
    "Project is hidden.",
    projectData?.projectSettings?.projectRef?.hidden === true, // condition to show
  );
```